src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -195,6 +195,32 @@ class ResourceHashtable : public ResourceObj {
       ++bucket;
     }
   }
+
+  // ITER contains bool do_entry(K const&, V const&), which will be
+  // called for each entry in the table.  If do_entry() returns true,
+  // the entry is deleted.
+  template<class ITER>
+  void unlink(ITER* iter) const {
+    Node** bucket =  const_cast<Node**>(_table);
+    while (bucket < &_table[SIZE]) {
```

Comment:
This is an odd way to iterate over a bucket array, I like the new version more. But I see you did it this way to stay consistent with the other methods.

Reply:
I could have taken bucket_at() wrapping the const_cast, too. But that calls 
table() which is not in 17, either. So I thought I only take this method and modify it.
Node ** is needed because below the buckets might be updated with the new list start.